dags/hindawi/parser.py

```diff
@@ -0,0 +1,201 @@
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='037']/ns0:subfield/[@code='9']",
+            self.prefixes,
+        )
+        if arxiv is None:
+            self.logger.error("No arxiv id_get_copyright_statement found.")
+            return None
+        if arxiv.text.lower() == "arxiv":
+            arxiv_value = article.find(
+                "ns0:metadata/ns1:record/ns0:datafield/[@tag='037']/ns0:subfield/[@code='a']",
+                self.prefixes,
+            ).text.lower()
+            return [{"value": arxiv_value.replace("arxiv:", "").strip()}]
+
+    def _get_journal_pages(self, article: ET.Element):
+        journal_pages = article.find(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='773']/ns0:subfield/[@code='c']",
+            self.prefixes,
+        ).text
+        if "-" in journal_pages:
+            return journal_pages.split("-", 1)
+        else:
+            return journal_pages, ""
+
+    def _construct_license(self, url, license_type, version):
+        if url and license_type and version:
+            return {
+                "url": url,
+                "license": f"CC-{license_type}-{version}",
+            }
+
+    def _get_license(self, article: ET.Element):
+        licenses = []
+        license_urls = article.findall(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='540']/ns0:subfield/[@code='u']",
+            self.prefixes,
+        )
+        license_texts = article.findall(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='540']/ns0:subfield/[@code='a']",
+            self.prefixes,
+        )
+        for license_url, license_text in zip(license_urls, license_texts):
+            if license_url.text:
+                url_parts = license_url.text.split("/")
+                clean_url_parts = list(filter(bool, url_parts))
+                version = clean_url_parts.pop()
+                licencse_type = clean_url_parts.pop()
+                licenses.append(
+                    self._construct_license(
+                        license_url.text, licencse_type.upper(), version
+                    )
+                )
+            elif license_text.text:
+                licenses.append(license_text.text)
+            return licenses
+
+
+class ExtractRights(HindawiParser):
+    def __init__(self):
+        self.statement = None
+
+    def _get_copyright(self, article):
+        copyright_statement = article.find(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='542']/ns0:subfield/[@code='f']",
+            self.prefixes,
+        ).text
+        self.statement = copyright_statement
+        return copyright_statement
+
+    def _get_copyright_statement(self, article):
+        if self.statement is None:
+            self.statement = self._get_copyright(article)
+        return self.statement
+
+    def _get_copyright_year(self, article):
+        copyright_year = re.search(r"[0-9]{4}", self.statement).group(0)
+        return copyright_year
```

Comment:
let's just add this logic to the parser class itself

Reply:
how? 😅 I did it with a separate class in order to avoid calling the function twice. Otherwise, we have to parse marcxml  two times